### PR TITLE
bug(spawn): refactor spawn to work with promise

### PR DESCRIPTION
**Pull request by jvcjunior**

### CODE REVIEW

File: utils/spawnHelper.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const spawn = require('cross-spawn')
+
+const spawnPromise = (command, args, options) => {
+    return new Promise ((resolve, reject) => {
+        const execution = spawn(command, args, options)
+
+        execution.on('close', (code) => {
+            resolve();
```
Comment (D40661195) [inline utils/spawnHelper.js]: I think it's better to pass code in resolve

File: utils/spawnHelper.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        });
+
+        execution.on('error', (err) => {
+            reject();
```
Comment (D40661195) [inline utils/spawnHelper.js]: I think it's better to pass err in resolve

**Review by italopessoa — APPROVED** [review]

File: awesomeCreateApp.js [hunk 3]
```diff
@@ -1,11 +1,9 @@
-const child_process = require('child-process-promise')
 const chalk = require('chalk')
 const inquirer = require('inquirer')
 const emoji = require('node-emoji')
 const hooks = require('./hooks')
+const spawnPromise = require('./utils/spawnHelper')
```
Comment (jouderianjr) [inline awesomeCreateApp.js]: maybe just `spawn` is better, all the async functions used in this context are using promises 😄

File: hooks/eslint/index.js [hunk 4]
```diff
@@ -1,10 +1,9 @@
-const child_process = require('child-process-promise')
 const fs = require('fs-extra')
 const packageJsonHelper = require('./../../utils/packageJsonHelper')
 const emoji = require('node-emoji')
 const chalk = require('chalk')
+const spawnPromise = require('./../../utils/spawnHelper')
```
Comment (jouderianjr) [inline hooks/eslint/index.js]: Same

File: hooks/eslint/index.js [hunk 5]
```diff
@@ -32,8 +31,11 @@ const addEslintFileSuccess = (cwd, folderName) => {
     stdio:'inherit'
   }
 
-  return spawn(command, [], terminalOpts)
+  return spawnPromise(command, [], terminalOpts)
     .then(() => installPackagesSuccess(cwd, folderName))
+    .catch((error) => {
```
Comment (jouderianjr) [inline hooks/eslint/index.js]: I think add error handling is not related with this particular PR, we could open an issue for that

File: hooks/eslint/index.js [hunk 6]
```diff
@@ -42,6 +44,8 @@ const installPackagesSuccess = (cwd, folderName) => {
       data.scripts.eslint =  'eslint .'
 
       return savePackageJsonIn(`${cwd}/${folderName}/package.json`, data)
+    }).catch(() => {
```
Comment (jouderianjr) [inline hooks/eslint/index.js]: I think add error handling is not related with this particular PR, we could open an issue for that

File: awesomeCreateApp.js [hunk 7]
```diff
@@ -15,8 +13,9 @@ const terminalOpts = {
 }
 
 const createApp = folderName => {
-  spawn('create-react-app', [folderName], terminalOpts)
-    .then( () => showQuestions(folderName))
+  spawnPromise('create-react-app', [folderName], terminalOpts)
+    .then(() => showQuestions(folderName))
+    .catch((error) => console.log('create app rejected', error))
```
Comment (jouderianjr) [inline awesomeCreateApp.js]: I think add error handling is not related with this particular PR, we could open an issue for that

**Review by jouderianjr — APPROVED** [review]

**Pull request merged**